components/core/src/ArrayBackedPosIntSet.hpp

```diff
@@ -0,0 +1,195 @@
+#ifndef ARRAYBACKEDPOSINTSET_HPP
+#define ARRAYBACKEDPOSINTSET_HPP
+
+// C++ standard libraries
+#include <unordered_set>
+#include <vector>
+
+// spdlog
+#include <spdlog/spdlog.h>
+
+// Project headers
+#include "Defs.h"
+#include "streaming_compression/zstd/Compressor.hpp"
+#include "TraceableException.hpp"
+
```

Comment:
Should have a header comment with a brief description of the class' purpose and the template parameter.